core/src/main/java/io/grpc/ResolvedServerInfoGroup.java

```diff
@@ -0,0 +1,109 @@
+ *  * met:
+ *  *
+ *  *    * Redistributions of source code must retain the above copyright
+ *  * notice, this list of conditions and the following disclaimer.
+ *  *    * Redistributions in binary form must reproduce the above
+ *  * copyright notice, this list of conditions and the following disclaimer
+ *  * in the documentation and/or other materials provided with the
+ *  * distribution.
+ *  *
+ *  *    * Neither the name of Google Inc. nor the names of its
+ *  * contributors may be used to endorse or promote products derived from
+ *  * this software without specific prior written permission.
+ *  *
+ *  * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ *  * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ *  * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ *  * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ *  * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ *  * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ *  * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ *  * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ *  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+package io.grpc;
+
+import com.google.common.collect.ImmutableList;
+
+import java.net.SocketAddress;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import javax.annotation.concurrent.Immutable;
+
+@ExperimentalApi("https://github.com/grpc/grpc-java/issues/1770")
+@Immutable
+public final class ResolvedServerInfoGroup {
+  private final List<ResolvedServerInfo> resolvedServerInfoList;
+  private final Attributes attributes;
+
+  public ResolvedServerInfoGroup(List<ResolvedServerInfo> resolvedServerInfoList, Attributes attributes) {
+    this.resolvedServerInfoList = resolvedServerInfoList;
+    this.attributes = attributes;
+  }
+
+  public ResolvedServerInfoGroup(List<ResolvedServerInfo> resolvedServerInfoList) {
+    this(resolvedServerInfoList, Attributes.EMPTY);
+  }
+
+  public ResolvedServerInfoGroup(ResolvedServerInfo resolvedServerInfo) {
+    this(Collections.singletonList(resolvedServerInfo));
+  }
+
+  public List<ResolvedServerInfo> getResolvedServerInfoList() {
+    return resolvedServerInfoList;
+  }
+
+  public Attributes getAttributes() {
+    return attributes;
+  }
+
+  public EquivalentAddressGroup toEquivalentAddressGroup() {
+    List<SocketAddress> addrs = new ArrayList<SocketAddress>(resolvedServerInfoList.size());
+    for (ResolvedServerInfo resolvedServerInfo : resolvedServerInfoList) {
+      addrs.add(resolvedServerInfo.getAddress());
+    }
+    return new EquivalentAddressGroup(addrs);
+  }
+
```

Comment:
Commented in your PR: https://github.com/grpc/grpc-java/pull/2175#discussion_r75684965